src/nvim/macros.h

```diff
@@ -133,12 +133,12 @@
 /* Get the length of the character p points to */
 # define MB_PTR2LEN(p)          (has_mbyte ? (*mb_ptr2len)(p) : 1)
 /* Advance multi-byte pointer, skip over composing chars. */
-# define mb_ptr_adv(p)      p += has_mbyte ? (*mb_ptr2len)(p) : 1
+# define mb_ptr_adv(p)      p += has_mbyte ? (*mb_ptr2len)((char_u *)p) : 1
 /* Advance multi-byte pointer, do not skip over composing chars. */
 # define mb_cptr_adv(p)     p += \
   enc_utf8 ? utf_ptr2len(p) : has_mbyte ? (*mb_ptr2len)(p) : 1
 /* Backup multi-byte pointer. Only use with "p" > "s" ! */
-# define mb_ptr_back(s, p)  p -= has_mbyte ? ((*mb_head_off)(s, p - 1) + 1) : 1
+# define mb_ptr_back(s, p)  p -= has_mbyte ? ((*mb_head_off)((char_u *)s, (char_u *)p - 1) + 1) : 1
```

Comment:
This line and line 136 above are the ones I'm most concerned about.  Doing it this way means any call to these defined functions will have the values cast as char_u.  mb_ptr2len() and mb_head_off() requires that anyway in theory, but I still feel like there should be a better way.